### ISSUE TITLE
bug: Angular APP_INITIALIZER waiting for platform.ready() is not working

### ISSUE THREAD

**Issue by Spinnenzunge** [ionic-team/ionic#18873]
# Bug Report
APP_INITIALIZER factory does not support initializing of cordova plugins.

**Ionic version:**

[x] **4.6.2**

**Current behavior:**
If I inject the `platform` from `@ionic/angular` in my `APP_INITIALIZER` factory, I get following error:

`"Could not find cordova.js script tag. Plugin loading may fail."`

I need to setup some code before routing in my Ionic 4 App. These initializing snippets have to access `cordova` plugins. (e.g. Storage and SecureStorage).

Therefore I need to wait for the `platform.ready()` event.

It looks like this is not possible because `APP_INITIALIZER` does not support the injection of `platform` and I get various side effects (e.g. Splashscreen hiding immediately, even if `AutoHideSplashScreen` setting is set to `false` in my `config.xml`)

**Expected behavior:**
Cordova plugins and platform can be injected in APP_INITIALIZER factory without side effects.

**Related code:**
app.module.ts
```
    {
      provide: APP_INITIALIZER,
      useFactory: AppService.factory,
      deps: [AppService],
      multi: true
    }
```

AppService:
```
@Injectable({
  providedIn: 'root'
})
export class AppService {

  constructor(private platform: Platform) {
  }

  static factory(appService: AppService) {
    return () => appService.initialize();
  }

  async initialize(): Promise<any> {
    await this.platform.ready();
  }
}
```

**Other information:**

**Ionic info:**

```
Ionic:

   ionic (Ionic CLI)             : 4.12.0 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.6.2
   @angular-devkit/build-angular : 0.13.9
   @angular-devkit/schematics    : 7.3.9
   @angular/cli                  : 7.3.9
   @ionic/angular-toolkit        : 1.5.1

Cordova:

   cordova (Cordova CLI) : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms     : android 8.0.0, ios 5.0.1
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 2.4.1, (and 19 other plugins)

System:

   Android SDK Tools : 26.1.1 (/Users/pascalgraf/Library/Android/sdk)
   ios-deploy        : 1.9.4
   ios-sim           : 8.0.1
   NodeJS            : v10.16.0 (/usr/local/bin/node)
   npm               : 6.10.0
   OS                : macOS Mojave
   Xcode             : Xcode 10.2.1 Build version 10E1001
```

**Comment by manucorporat** [comment]
APP_INITIALIZER Is not an ionic api, but an angular one. How are we supposed to fix this issue within ionic?

**Comment by Spinnenzunge** [comment]
Hey @manucorporat I just did some debugging the last billion hours and I found out that my side effects with the splash screen are caused by the cordova-secure-storage plugin in conjunction with the splash-screen plugin. The secure storage plugin pauses the cordova activity and resumes it again after:
```
D/CordovaActivity: Paused the activity.
D/CordovaActivity: Resumed the activity.
```

This pause triggers the splashscreen hide() method: https://github.com/Crypho/cordova-plugin-secure-storage/issues/138

For this Issue: I opened it, because I thought maybe the platform from @ionic/angular could be edited to support the Injection to an APP_INITIALIZER factory but it seems my problem was a completely different one.

I will close this Issue, nonetheless I'm still not sure how well angular APP_INITIALIZER works in an App running Ionic and Cordova.
